frontend/src/utils/StudentUtils.ts

```diff
@@ -0,0 +1,57 @@
+import update from "immutability-helper";
+
+import { QuestionComponentRequest } from "../APIClients/types/TestClientTypes";
+import { Answers } from "../types/AnswerTypes";
+import { ResponseElementType } from "../types/QuestionTypes";
+
+export const getAnswerElements = (
+  question: QuestionComponentRequest[],
+): QuestionComponentRequest[] => {
+  return question.filter(
+    (questionElement) => questionElement.type in ResponseElementType,
+  );
+};
+
+export const getAnswerValues = (
+  currentQuestionIndex: number,
+  answerElementIndex: number,
+  answers: Answers[],
+): number[] => {
+  const answer = answers.find((a) => a.index === currentQuestionIndex);
```

Comment:
Will the index of an answer ever change from the initialization? If not, might be a bit more efficient to construct (once) a dict from elementIndex -> answerIndex and use that, especially since (I think) this code runs on every answer change

Reply:
wait. can i not just simplify this to -> `const answer = answers[currentQuestionIndex]` and get rid of all the `index` fields from the `Answers` type? please sanity check me on this haha